internals/plan/extensions_test.go

```diff
@@ -0,0 +1,687 @@
+// Copyright (c) 2024 Canonical Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plan_test
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	. "gopkg.in/check.v1"
+	"gopkg.in/yaml.v3"
+
+	"github.com/canonical/pebble/internals/plan"
+)
+
+type inputLayer struct {
+	order int
+	label string
+	yaml  string
+}
+
+// PlanResult represents the final content of a combined plan. Since this
+// package exclusively focuses extensions, all built-in sections are empty
+// and ignored in the test results.
+type planResult struct {
+	x *xSection
+	y *ySection
+}
+
+var extensionTests = []struct {
+	summary    string
+	extensions map[string]plan.LayerSectionExtension
+	layers     []*inputLayer
+	error      string
+	result     *planResult
+	resultYaml string
+}{
+	{
```

Comment:
There's an unnecessary indendation on every line below:
```
}{{
        summary: ...
}, {
    ...
}}

Reply:
Fixed, thank you. Resolving.